### PR TITLE
nimble/ll: Randomize initial advertising event

### PR DESCRIPTION
**Pull request by sjanc**
When starting advertising we can also randomize first event. This is
primary useful for mesh relay where multiple devices starts advertising
at the same time.

This is a bit of RFC :)

Also I don't think we need 5ms fixed delay anymore but I've left it as is for now...

### CODE REVIEW

**Comment by sjanc** [comment]
@haukepetersen  FYI

File: nimble/controller/src/ble_ll_adv.c [hunk 1]
```diff
@@ -2596,16 +2597,18 @@ ble_ll_adv_sm_start(struct ble_ll_adv_sm *advsm)
...
-     * to check if we are currently doing anything. Just makes this simple.
+     * advertising event some time in the future (5 msecs + adv rand). This will
+     * give time to start up any clocks or anything and also avoid a bunch of
+     * code to check if we are currently doing anything. Just makes this simple.
      *
      * Might also want to align this on a slot in the future.
      *
      * NOTE: adv_event_start_time gets set by the sched_adv_new
      */
+    start_delay_us = 5000 + (rand() % (BLE_LL_ADV_DELAY_MS_MAX * 1000));
```
Comment (rymanluk) [inline nimble/controller/src/ble_ll_adv.c]: maybe instead `rand()` we could use `ble_ll_rand_data_get()`?

Comment (andrzej-kaczmarek) [reply nimble/controller/src/ble_ll_adv.c]: we need random number, not random data...

**Comment by haukepetersen** [comment]
Nice, looks good to me!

**Pull request merged**